crates/router/src/core/payments.rs

```diff
@@ -4071,11 +4081,18 @@ where
             .unwrap_or_default();
         (profile_id, algorithm_ref.algorithm_id)
     };
+    // Validate that the profile_id being passed in intent should be same as the profile_id passed from the business profile parameter
+    r_utils::when(&business_profile.profile_id != profile_id, || {
```

Comment:
we already validate this and only then payment intent is created, so this check should not be required